pkg/scheduler/util/utils.go

```diff
@@ -140,3 +140,32 @@ func MoreImportantPod(pod1, pod2 interface{}) bool {
 	}
 	return GetPodStartTime(pod1.(*v1.Pod)).Before(GetPodStartTime(pod2.(*v1.Pod)))
 }
+
+// NonBlockingSendError sends val to the provided channel without blocking on the receiver.
+// Note that this function should only be used for errors since type casting the interface
+// adds overhead that is best avoided in performance critical paths.
+func NonBlockingSendError(ch chan<- interface{}, val interface{}) {
```

Comment:
I am fine with these functions, but please do not add them to this file. @misterikkit tries hard to remove many things from this file. We should try to shrink and eventually remove it. You can move these to a new file of its own.

I think you should also create a struct that has a channel member and these functions are its methods. A "New" method can be used to create the struct and these functions do not need to receive the channel every time they are called. Something like:

```go
type ErrorChannel struct {
  ch chan<- interface{}
}

func (e *ErrorChannel) SendError(val interface{}) {
...
}

func (e *ErrorChannel) SendErrorWithCancel(val interface{}, cancel context.CancelFunc) {
...
}

func (e *ErrorChannel) ReceiveError() ...

func NewErrorChannel() *ErrorChannel {
...
}
```

Reply:
ok, changed it back to simple lock, and will use the new interface once merged and when we do the NormalizeScore part.